as.c

```diff
@@ -5,6 +5,138 @@
 #include "vm.h"
 #include "opcode.h"
 
+#define NO_QUOTE '\0'
+#define IS_QUOTE(qs, c) ((qs == NO_QUOTE) ? (c == '"' || c == '\'') : c == qs)
+#define SET_QUOTE(qs, c) \
+    do {                 \
+        qs = c;          \
+    } while (0)
+#define IS_ESC(c) (c == '\\')
+
+static char *quoted_strsep(char **str, const char *sep)
```

Comment:
Since it is a customized `strseq` implementation, appropriate comment(s) should exist as well.